frontend/src/components/maps/leaflet/LayerPopup/hooks/useLayerQuery.tsx

```diff
@@ -126,13 +140,29 @@ export const useLayerQuery = (url: string): IUserLayerQuery => {
     [baseUrl],
   );
 
+  const findMetadataByLocation = useCallback(
+    async (
+      latlng: LatLngLiteral,
+      geometryName: string = 'SHAPE',
+      spatialReferenceId: number = 4326,
+    ): Promise<Record<string, any>> => {
+      const collection = await findOneWhereContains(latlng, geometryName, spatialReferenceId);
+      if (collection?.features?.length > 0) {
+        return collection.features[0].properties || {};
```

Comment:
would it not be better to just return undefined instead of an empty object?

Reply:
In my opinion an empty object here is okay as it is saying "there is no metadata". That way when you ask for obj.SOME_FIELD you don't have to check for undefined first. 

Again personal preference. The method signature is that it returns a `Record<string,any>` not `Record | undefined`